app/models/ems_refresh/save_inventory_networks.rb

```diff
@@ -153,12 +153,12 @@ def save_network_routers_inventory(ems, hashes, target = nil)
       store_ids_for_new_records(ems.network_routers, hashes, :ems_ref)
     end
 
-    def save_network_ports_inventory(ems, hashes, target = nil)
+    def save_network_ports_inventory(ems, hashes, target = nil, source = :neutron)
```

Comment:
Note that in those other places we don't record the source, so that bit is different.  Instead we prefer one over the other to avoid conflicts.

On that note, I am concerned about duplication of data.  You could get a network_port in there twice, one from each source, but that might cause problems with reporting, particularly with counts.